testsrc/main/mondrian/test/NativeSetEvaluationTest.java

```diff
@@ -1647,6 +1662,140 @@ public void testNativeSetsCacheClearing() {
                 getTestContext(), mdx, patterns, false, false, false);
         }
     }
+
+    /**
+    * tests if the results of native and non-native evaluations are equal
+    * and if exception is thrown in case of native evaluation and
+    * mondrian.native.unsupported.alert equals to "ERROR".
+    *
+    * @see <a href="http://jira.pentaho.com/browse/MONDRIAN-1803">Jira issue</a>
+    */
+    public void testNativeFilterWithLargeAggSetInSlicer() {
+      final String query = "with member customers.agg as "
+          + "'Aggregate(Except(Customers.[Name].members,    "
+          + "{[Customers].[USA].[OR].[Corvallis].[Judy Doolittle]}    ))' "
+          + " select filter(gender.gender.members, measures.[unit sales] >131500)"
+          + " on 0 from sales "
+          + " where customers.agg";
+      final String message =
+          "The results of native and non-native evaluations should be equal";
+      verifySameNativeAndNot(query, message, getTestContext());
+
+      propSaver.set(
+          MondrianProperties.instance()
+              .AlertNativeEvaluationUnsupported, "ERROR");
+      propSaver.set(propSaver.properties.EnableNativeFilter, true);
+      Appender appenderMock = mock(Appender.class);
+      try {
+        final Logger logger = Logger.getLogger(RolapUtil.class);
+        logger.addAppender(appenderMock);
+        executeQuery(query);
+      } catch (NativeEvaluationUnsupportedException e) {
+        // expected behavior: verify log message and return
+        verify(appenderMock)
+            .doAppend(argThat(
+                new ArgumentMatcher<LoggingEvent>() {
+          @Override
+          public boolean matches(Object item) {
+            String message =
+                "Unable to use native SQL evaluation for 'Filter'; reason: "
+                + MondrianResource.instance()
+                .MembersQuantityExceedsMaxConstraints.str("10280");
+            return message.equals(
+                ((LoggingEvent) item).getMessage().toString());
+          }
+        }));
+        return;
+      }
+      fail("NativeEvaluationUnsupportedException was expected");
+    }
+
+  /**
+   * the sum of calculated members for [Customers].[Country].Members is 3,
+   * for [Store].[Store State].Members is 10.
+   * At first we set MaxConstraints to 11
+   * (exceeds values of calculated members for each aggs).
+   * Here we verify that the generated SQL contains "in" clause
+   * and no ERROR messages are occurring.
+   * After that we assign MaxConstraints to 9
+   * (less than the sum of calc members for
+   * [Store].[Store State].Members) and verify that
+   * exception was thrown and log contains an error message. Beforehand we set
+   * AlertNativeEvaluationUnsupported to ERROR.
+   * @see <a href="http://jira.pentaho.com/browse/MONDRIAN-1803">Jira issue</a>
+   */
+  public void testNativeFilterWithLargeAggSetInSlicerTwoAggs() {
+    String query =
```

Comment:
This test will fail with aggregates enabled.  
In other places we've just added a check for whether aggs are enabled at the top of the test and aborted.  Alternatively you could add the aggregate SQL